esda/join_counts.py

```diff
@@ -150,12 +152,16 @@ def __init__(self, y, w, permutations=PERMUTATIONS, drop_islands=True):
         y = np.asarray(y).flatten()
         w.transformation = "b"  # ensure we have binary weights
         self.w = w
-        self.adj_list = self.w.to_adjlist(
-            remove_symmetric=False, drop_islands=drop_islands
-        )
+        if isinstance(w, W):
+            self.adj_list = self.w.to_adjlist(
+                remove_symmetric=False, drop_islands=drop_islands
+            )
+        else:
+            self.adj_list = self.w.adjacency.reset_index()
         self.y = y
         self.permutations = permutations
-        self.J = w.s0 / 2.0
+        s0 = w._s0 if hasattr(w, "_s0") else w.s0
```

Comment:
I'm not clear on why these need to be private (s0 in this case)? What is the advantage of that?

Reply:
@ljwolf didn't like exposing them as public properties. I originally mirrored weights API which was easy and I'm happy to follow whichever decision we make in lib.